test/e2e/singlecluster/pod_test.go

```diff
@@ -217,7 +217,7 @@ var _ = ginkgo.Describe("Pod groups", func() {
 					var p corev1.Pod
 					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(group[0]), &p)).To(gomega.Succeed())
 					return p.Status.Phase
-				}, util.Timeout, util.Interval).Should(gomega.Equal(corev1.PodFailed))
+				}, util.LongTimeout, util.Interval).Should(gomega.Equal(corev1.PodFailed))
```

Comment:
I suspect what happens is that we may occasionally hit with SIGTERM the pod before the SIGTERM trap is registered, and then it takes by default 30s before sending SIGKILL. 
Maybe we can try first setting for this case `TerminationGracePeriod(1).` on the PodSpec and use longer timeout only if this repeats.

I agree it is not responsibility of Kueue to test kubelet, but we should also try to keep the tests executing fast, and extending the timeout may cover some possible optimizations.

Reply:
TerminationGracePeriod(1) helped a little bit but still have this error.

Reply:
This is surprising. How do you tell? Did you run the test locally in a loop and it still failed?

Reply:
Yes, I reduced resources and run test with `--until-it-fails`.

Reply:
I see, thank you for doing so. As you have the reproducible setup two asks to better understand the nature of the issue:
1. How often,roughly, does it happen
2. Could you check if waiting for the pod to be Running before deleting + using terminationGracwPeeiod makes it stable?

Reply:
1. It's about 1 of 10 is failed.
2. Yes, you right, checking for Running makes it stable.

Reply:
I see, thank you for the research. I don't have a strong view here. On one hand the test would run slightly faster and more predictable, on the other there is some complexity and extra code to wait for running.

I would be slightly leaning to wait for running + use terminatingGracefulPeriod in that case, so that we can compensate the complexity with a STEP comment, for example `ginkgo.By("Await for the Pod to be running to allow fast termination by Kubelet")`.

WDYT?

Reply:
Yes, totally agree with you.